nestkernel/nest_types.h

```diff
@@ -32,6 +32,19 @@
 // Generated includes:
 #include "config.h"
 
+#ifdef _OPENMP
+  #define NEST_PARALLEL_FOR_THREAD(t) \
+    _Pragma("omp parallel") \
+    { \
+      size_t t = kernel().vp_manager.get_thread_id();
```

Comment:
shouldn't the type be `thread`?
I guess you already tried making this `const`, but this probably breaks somewhere in `connect`. Are we sure this does not affect performance? Or am I misunderstanding how macros work?! Could we create two macros, one with and one without const?